2018-07-14 16:53:11,103 INFO: Microblog startup [in D:\python_projects\flask\app\__init__.py:76]
2018-07-14 17:51:08,097 INFO: Microblog startup [in D:\python_projects\flask\app\__init__.py:76]
2018-07-14 17:51:11,496 ERROR: Exception on /auth/login [GET] [in d:\python_projects\flask\venv\lib\site-packages\flask\app.py:1761]
Traceback (most recent call last):
  File "d:\python_projects\flask\venv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "d:\python_projects\flask\venv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "d:\python_projects\flask\venv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "d:\python_projects\flask\venv\lib\site-packages\flask\_compat.py", line 35, in reraise
    raise value
  File "d:\python_projects\flask\venv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "d:\python_projects\flask\venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "D:\python_projects\flask\app\auth\routes.py", line 26, in login
    return render_template('auth/login.html', title='Sign in', form=form)
  File "d:\python_projects\flask\venv\lib\site-packages\flask\templating.py", line 135, in render_template
    context, ctx.app)
  File "d:\python_projects\flask\venv\lib\site-packages\flask\templating.py", line 117, in _render
    rv = template.render(context)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "D:\python_projects\flask\app\templates\base.html", line 41, in template
    {% endblock %}
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\environment.py", line 1005, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "D:\python_projects\flask\app\templates\auth\login.html", line 1, in top-level template code
    {% extends 'base.html' %}
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "D:\python_projects\flask\app\templates\base.html", line 41, in template
    {% endblock %}
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 268, in parse_block
    node.body = self.parse_statements(('name:endblock',), drop_needle=True)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 213, in parse_if
    'name:endif'))
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 144, in parse_statement
    self.fail_unknown_tag(token.value, token.lineno)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 97, in fail_unknown_tag
    return self._fail_ut_eof(name, self._end_token_stack, lineno)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 90, in _fail_ut_eof
    self.fail(' '.join(message), lineno)
  File "d:\python_projects\flask\venv\lib\site-packages\jinja2\parser.py", line 59, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endblock'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'elif' or 'else' or 'endif'. The innermost block that needs to be closed is 'if'.
